demisto_sdk/commands/validate/validators/RM_validators/RM108_is_image_path_valid.py

```diff
@@ -72,20 +72,27 @@ def detect_absolute_image_paths(self, content_item) -> str:
             re.IGNORECASE | re.MULTILINE,
         )
         if absolute_links:
-            # Extract the actual link from the found absolute links,
-            absolute_links = [
-                # extracts the URL from the matched link tuple, choosing between absolute_link[1] or absolute_link[2]
-                # based on whether the link matches the URL or HTML pattern.
-                absolute_link[1] if absolute_link[0] else absolute_link[2]
+            # Extract and filter links in a single pass
+            absolute_links_without_assets = [
+                # Extract the correct link and filter out unwanted assets in one step
+                (absolute_link[1] if absolute_link[0] else absolute_link[2])
                 for absolute_link in absolute_links
+                if not (
+                    "content-assets"
+                    in (absolute_link[1] if absolute_link[0] else absolute_link[2])
+                    and (
+                        absolute_link[1] if absolute_link[0] else absolute_link[2]
+                    ).endswith(".gif")
+                )
```

Comment:
Let's refactor this code. It is very unclear.
Also, why store the link in a tuple?

Reply:
Tuples are generally faster and more memory-efficient, plus I don't need to make changes to the items.
https://www.simplilearn.com/difference-between-list-and-tuple-in-python-article#:~:text=Tuples%20are%20generally%20faster%20and,when%20iteration%20speed%20is%20crucial.